test/unit/test_rank_unit.py

```diff
@@ -0,0 +1,297 @@
+"""Unit tests for the rank command."""
+
+from unittest import mock
+
+from util import get_mock_State_and_config
+
+from wily.commands.rank import rank
+
+EXPECTED = """
+╒════════╤═════════════════╕
+│ File   │   Lines of Code │
+╞════════╪═════════════════╡
+│ file1  │               0 │
+├────────┼─────────────────┤
+│ file2  │               1 │
+├────────┼─────────────────┤
+│ Total  │               1 │
+╘════════╧═════════════════╛
+"""
+EXPECTED = EXPECTED[1:]
+
+
+def test_rank(capsys):
+    metric = "raw.loc"
+    revision_id = "abcdeff"
+    mock_State, mock_config = get_mock_State_and_config(3, ascending=True)
+    mock_revision = mock.MagicMock(key="abcdeff123123", message="Nothing.")
+    mock_resolve = mock.MagicMock()
+    mock_resolve.cls.find = mock.Mock(return_value=mock_revision)
+
+    with mock.patch("wily.commands.rank.State", mock_State), mock.patch(
+        "wily.commands.rank.resolve_archiver", mock_resolve
+    ):
+        rank(
+            config=mock_config,
+            path=None,
```

Comment:
path, limit and threshold are typed as being required arguments, so this test shouldn't even pass :-)

Reply:
Done. `path` can indeed be `None`, so I've updated the typing annotation to `Optional[str]`. That happens because in `__main__.py` we have:
```python
@click.argument("path", type=click.Path(resolve_path=False), required=False)
```

The wrong typing on the test only "worked" because the actual call doesn't check for typing conformance and we don't run pyright on test files. I'll submit another PR that fixes the few other mistyped tests and enables pyright on the test directory.